### PR TITLE
Add support for allowing cert-manager signers

### PR DESCRIPTION
**Pull request by hazim1093**
In order to use [Custom CA Integration using Kubernetes CSR](https://istio.io/latest/docs/tasks/security/cert-management/custom-ca-k8s), we need to modify the Istiod ClusterRole and add a cert-manager `issuer` or `clusterissuer` in the list of resources allowed to `approve` signers. 

In the [guide](https://istio.io/latest/docs/tasks/security/cert-management/custom-ca-k8s/#deploy-istio-with-default-cert-signer-info), this is supported by being able to modify the ClusterRole via`overlays`.

### CODE REVIEW

**Comment by hazim1093** [comment]
Hi @GregHanson, 

so the tests fail as `certSigners` is not part of v1alpha1.GlobalConfig. 
Should this belong somewhere else then?

**Comment by aimuz** [comment]
You may also need to modify this issue for generating a new pb.go

 operator/pkg/apis/istio/v1alpha1/values_types.proto

**Comment by hazim1093** [comment]
@aimuz not for now at least. Right now it just adds resources in the clusterrole to allow them, when using cert-manager for mTLS CA

**Comment by aimuz** [comment]
operator/pkg/apis/istio/v1alpha1/values_types.proto

Modify this file so that it supports this field

**Comment by aimuz** [comment]
Please sign the CLA

**Comment by aimuz** [comment]
You may need to execute this command

```
make gen
```

If you can, can you combine the commit records into one via git rebase?

File: manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml [hunk 1]
```diff
@@ -82,6 +82,9 @@ rules:
       - "signers"
     resourceNames:
     - "kubernetes.io/legacy-unknown"
+{{- range .Values.global.certSigners }}
+    - {{ . | quote }}
+{{- end }}
```
Comment (hzxuzhonghu) [inline manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:87]: I donot understand the whole bunch, IIUC k8s does not have `signers` resources at all

Comment (hzxuzhonghu) [reply manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:87]: So i doubt this is useless

Comment (hzxuzhonghu) [reply manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:87]: IC, the certmanager has a groups `"cert-manager.io"`, which has the `signers` resource, but quite different from `certificates.k8s.io` group.

Comment (hzxuzhonghu) [reply manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:87]: This bunch is not correct before and after

Comment (hazim1093) [reply manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:87]: It is part of the [kubenetes CSR API](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#signers/) 

And it is the same group in the [istio guide](https://istio.io/latest/docs/tasks/security/cert-management/custom-ca-k8s/#deploy-istio-with-default-cert-signer-info) that uses CSR API

Comment (hzxuzhonghu) [reply manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:87]: OK. never mind. CSR is special

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 2]
```diff
@@ -215,6 +215,11 @@ meshConfig:
 global:
   # Used to locate istiod.
   istioNamespace: istio-system
+  # List of cert-signers to allow "approve" action in the istio cluster role                                                                                                                                                           ─╯
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/values.yaml:218]:
```suggestion
  # List of cert-signers to allow "approve" action in the istio cluster role
```

File: manifests/charts/istiod-remote/values.yaml [hunk 3]
```diff
@@ -186,6 +186,11 @@ meshConfig:
 global:
   # Used to locate istiod.
   istioNamespace: istio-system
+  # List of cert-signers to allow "approve" action in the istio cluster role                                                                                                                                                           ─╯
```
Comment (howardjohn) [inline manifests/charts/istiod-remote/values.yaml:189]:
```suggestion
  # List of cert-signers to allow "approve" action in the istio cluster role
```

**Review by howardjohn — APPROVED** [review]

**Review by morvencao — COMMENTED** [review]
need to run `make gen` to fix the testing.

**Review by costinm — COMMENTED** [review]
I wonder if this is the right approach ( using values to add permissions ). 

Doesn't cert manager define a cluster role on their install template ?  Should we instead have a list of cluster roles to bind to ? 

I think bloated cluster roles with tons of unrelated permissions are hard to understand - it is easier to review and understand smaller specialized roles, and manage the bindings.

An extra benefit is that it decouples to some extent the istio install - bindings can be created/managed in an independent yaml, so users can integrate with different signers ( and other features requiring extra permissions ) without having to reinstall Istio. The pattern would be:

- user has an existing istio install ( may be a managed one - where they don't have control over the setup options)
-  they install certmanager, which creates the cluster role
- they apply a yaml binding the istiod SA to the cluster role - without touching istio install.

**Comment by costinm** [comment]
My comment above is a bit more generic and is not intended to block this PR - we use this pattern of bloated cluster role already. 

But we should consider a refactoring and simplification of the install, to minimize the number of 'install' options and support things like granting additional permissions for integrations independent of the install.

**Comment by hazim1093** [comment]
Hi, 

In the light of the previous discussion, is this good to be merged now?

**Review by hzxuzhonghu — APPROVED** [review]

**Review by costinm — APPROVED** [review]

**Pull request merged**